enterprise_access/apps/bffs/tests/utils.py

```diff
@@ -34,6 +37,20 @@ def setUp(self):
         }
         self.request.data = {}
 
+        # Mock HandlerContext
+        self.mock_handler_context = mock.MagicMock(
+            _request=self.request,
+            _status_code=status.HTTP_200_OK,
+            _errors=[],
+            _warnings=[],
+            _enterprise_customer_uuid=self.mock_enterprise_customer_uuid,
+            _enterprise_customer_slug=self.mock_enterprise_customer_slug,
+            _lms_user_id=self.request.user.lms_user_id,
+            _enterprise_features={},
+            _data={},
+        )
+        self.override_handler_context_getter_properties()
```

Comment:
[suggestion] It may make sense to have the helper method be responsible for creating the `mock.MagicMock`, too? E.g., maybe something like this:

```python
def get_mock_handler_context(self, **kwargs):
    """
    Returns a mock handler context with optional overrides.
    Any attribute in the context can be overridden by passing it as a keyword argument.
    """
    # Create default values for the mock HandlerContext
    default_values = {
        '_request': self.request,
        '_status_code': status.HTTP_200_OK,
        '_errors': [],
        '_warnings': [],
        '_enterprise_customer_uuid': self.mock_enterprise_customer_uuid,
        '_enterprise_customer_slug': self.mock_enterprise_customer_slug,
        '_lms_user_id': self.request.user.lms_user_id,
        '_enterprise_features': {'feature_a': True},
        'data': {},
    }

    # Update default values with any overrides provided via kwargs
    default_values.update(kwargs)

    # Create the mock HandlerContext
    mock_handler_context = mock.MagicMock(**default_values)

    # Define a dictionary of private attributes to property names
    property_mocks = {
        'request': getattr(mock_handler_context, '_request'),
        'status_code': getattr(mock_handler_context, '_status_code'),
        'errors': getattr(mock_handler_context, '_errors'),
        'warnings': getattr(mock_handler_context, '_warnings'),
        'enterprise_customer_uuid': getattr(mock_handler_context, '_enterprise_customer_uuid'),
        'enterprise_customer_slug': getattr(mock_handler_context, '_enterprise_customer_slug'),
        'lms_user_id': getattr(mock_handler_context, '_lms_user_id'),
        'enterprise_features': getattr(mock_handler_context, '_enterprise_features'),
    }

    # Override the property getters
    for property_name, private_attr in property_mocks.items():
        setattr(
            type(mock_handler_context),  # The class of the mock object
            property_name,
            mock.PropertyMock(return_value=private_attr)
        )

    return mock_handler_context
```

This change would help simplify/negate the need to call `override_handler_context_getter_properties` within `test_response_builders` after overriding `_status_code` on the mocked `HandlerContext` instance.

For example:

```python
@ddt.data(
    {'status_code': status.HTTP_400_BAD_REQUEST},
    {'status_code': None}
)
@mock.patch('enterprise_access.apps.bffs.context.HandlerContext')
@ddt.unpack
def test_get_status_code(self, mock_handler_context, status_code):
    if status_code:
        mock_handler_context.return_value = self.get_mock_handler_context(
            _status_code=status_code
        )
    else:
        mock_handler_context.return_value = self.mock_handler_context

    mock_handler_context_instance = mock_handler_context.return_value
    expected_output = status_code if status_code else status.HTTP_200_OK

    base_response_builder = BaseResponseBuilder(mock_handler_context_instance)
    response_status_code = base_response_builder.get_status_code()

    self.assertEqual(response_status_code, expected_output)
```